apps/api/src/domain/decorators/aggregate-root.decorator.ts

```diff
@@ -0,0 +1,27 @@
+import { Maybe } from '../../lib/types/maybe';
+import { NotFound } from '../../lib/types/not-found';
+import { isNullOrUndefined } from '../utilities/validation/is-null-or-undefined';
+
+const AGGREGATE_TYPE_METADATA = '__AGGREGATE_ROOT_METADATA__';
+
+export type AggregateTypeMetadata = {
+    aggregateType: string;
+};
+
+export const getAggregateTypeForTarget = (target: Object): Maybe<AggregateTypeMetadata> => {
+    const meta = Reflect.getMetadata(AGGREGATE_TYPE_METADATA, target);
+
+    if (isNullOrUndefined(meta)) return NotFound;
+
+    return meta as AggregateTypeMetadata;
+};
+
+export function AggregateRoot(aggregateType: string): ClassDecorator {
```

Comment:
Add a detailed comment about the significance of an `AggregateRoot` within COSCRAD.

Reply:
Done.